pulp_ansible/app/serializers.py

```diff
@@ -752,6 +906,64 @@ class Meta:
         )
 
 
+class CollectionVersionSigstoreSignatureSerializer(NoArtifactContentUploadSerializer):
+    """
+    A serializer for Sigstore signature models.
+    """
+
+    data = serializers.CharField(
+        help_text=_("A signature, base64 encoded."),
+        required=True,
+    )
+    sigstore_x509_certificate = serializers.CharField(
+        help_text=_("The ephemeral PEM-encoded signing certificate generated by Sigstore."),
+        required=True,
+    )
+    sigstore_bundle = serializers.JSONField(
+        help_text=_("A Sigstore bundle used for offline verification."),
+        allow_null=True,
+    )
+    signed_collection = DetailRelatedField(
+        help_text=_("The content this signature is pointing to."),
+        view_name_pattern=r"content(-.*/.*)-detail",
+        queryset=CollectionVersion.objects.all(),
+    )
+    sigstore_signing_service = RelatedField(
+        help_text=_("The signing service used to sign the collections"),
+        view_name="sigstore-signing-services-detail",
+        read_only=True,
+        allow_null=True,
+    )
+
+    def __init__(self, *args, **kwargs):
+        """Ensure `file` field is required."""
+        super().__init__(*args, **kwargs)
+        self.fields["file"].required = True
+
+    def validate(self, data):
+        """
+        Verify the signature is valid before creating it.
+        """
+        data = super().validate(data)
+
```

Comment:
Looking at `verify_sigstore_signature_upload` below I think we should add a check here to make sure the `repository` field is set and if not raise a `serializer.ValidationError` so that we error quicker.

Reply:
Addressed in https://github.com/pulp/pulp_ansible/pull/1428/commits/de1e8e1f031badfe015d5b421ba5e474631bf41c